### PR TITLE
Fix WNCLIENT log ownership

### PR DESCRIPTION
**Pull request by brianhlin**

### CODE REVIEW

File: 25-hosted-ce-setup.sh [hunk 1]
```diff
@@ -21,11 +21,10 @@ popd
 users=$(cat /etc/grid-security/grid-mapfile /etc/grid-security/voms-mapfile | \
             awk '/^"[^"]+" +[a-zA-Z0-9\-\._]+$/ {print $NF}' | \
             sort -u | \
-            tr '\n' ',')
+            tr '\n' ' ')
```
Comment (matyasselmeci) [inline 25-hosted-ce-setup.sh]: The user list in the sudoers file needs to be comma separated
```suggestion
            tr '\n' ',')
```

File: 25-hosted-ce-setup.sh [hunk 2]
```diff
@@ -21,11 +21,10 @@ popd
...
 [[ -n $users ]] || { echo >&2 "No users found in /etc/grid-security/grid-mapfile or /etc/grid-security/voms-mapfile"; exit 1; }
-groupadd boscouser
-gpasswd -M "${users}" boscouser
-echo "condor ALL = ( :boscouser ) NOPASSWD: /usr/bin/update-remote-wn-client" \
+# Use param expansion to remove the trailing space
+echo "condor ALL = (${users%%[[:space:]]}) NOPASSWD: /usr/bin/update-remote-wn-client" \
```
Comment (matyasselmeci) [inline 25-hosted-ce-setup.sh]:
```suggestion
# Use param expansion to remove the trailing comma
echo "condor ALL = (${users%%,}) NOPASSWD: /usr/bin/update-remote-wn-client" \
```

File: 25-hosted-ce-setup.sh [hunk 3]
```diff
@@ -21,11 +21,10 @@ popd
 users=$(cat /etc/grid-security/grid-mapfile /etc/grid-security/voms-mapfile | \
             awk '/^"[^"]+" +[a-zA-Z0-9\-\._]+$/ {print $NF}' | \
             sort -u | \
-            tr '\n' ',')
+            tr '\n' ' ')
 [[ -n $users ]] || { echo >&2 "No users found in /etc/grid-security/grid-mapfile or /etc/grid-security/voms-mapfile"; exit 1; }
```
Comment (matyasselmeci) [inline 25-hosted-ce-setup.sh]: Won't this always be true if there's a trailing space/comma in $users?

Comment (brianhlin) [reply 25-hosted-ce-setup.sh]: If there are no users, the pipeline will result in the empty string:

```
[root@osg-hosted-ce-blin-test-ce-85f745c7cd-pm6rm ~]# cat grid-mapfile voms-mapfile
[root@osg-hosted-ce-blin-test-ce-85f745c7cd-pm6rm ~]# users=$(cat grid-mapfile voms-mapfile | \
>             awk '/^"[^"]+" +[a-zA-Z0-9\-\._]+$/ {print $NF}' | \
>             sort -u | \
>             tr '\n' ' ')
[root@osg-hosted-ce-blin-test-ce-85f745c7cd-pm6rm ~]# [[ -n $users ]] || { echo >&2 "No users found in /etc/grid-security/grid-mapfile or /etc/grid-security/voms-mapfile"; } 
No users found in /etc/grid-security/grid-mapfile or /etc/grid-security/voms-mapfile
```

**Pull request merged**